src/mapgen.cpp

```diff
@@ -6311,8 +6311,43 @@ std::vector<item *> map::place_items(
             tries++;
         } while( is_valid_terrain( p ) && tries < 20 );
         if( tries < 20 ) {
-            auto put = put_items_from_loc( group_id, p, turn );
-            res.insert( res.end(), put.begin(), put.end() );
+            for( const item &itm : item_group::items_from( group_id, turn, spawn_flags::use_spawn_rate ) ) {
+                const float item_cat_spawn_rate = std::max( 0.0f, item_category_spawn_rate( itm ) );
+                // for items with category spawn rate less than or equal to 1, roll a dice to see if they should spawn
+                if( item_cat_spawn_rate <= 1 ) {
+                    if( rng_float( 0.1f, 1 ) <= item_cat_spawn_rate ) {
```

Comment:
Looks like if you set item_cat_spawn_rate too low for some category (less than 0.1f) the item will always spawn? Because this rng_float will always roll greater than 0.1f.